### PR TITLE
[YUNIKORN-852] Document resource weighting in node sorting policies.

### PR DESCRIPTION
**Pull request by craigcondit**

### CODE REVIEW

**Comment by craigcondit** [comment]
@yangwwei, can you review this?

File: docs/user_guide/sorting_policies.md [hunk 1]
```diff
@@ -146,6 +146,33 @@ Resources of the same type are compared for the nodes.
 This results in a node with the highest utilisation to be considered first for assigning new allocation.
 Resulting in a high(er) utilisation of a small(er) number of nodes, better suited for cloud deployments.   
 
+## Resource weighting
+Node sorting policies may use the utilization of a node to determine ordering. Because nodes can have several unique
+resource types, a node's utilization is determined by a weighted average of its individual resource types. Resource
+weighting can be customized by using the `resourceweights` section of `nodesortpolicy`. If `resourceweights` is not
+present or empty, the default configuration sets the weight of both `vcore` and `memory` equally to `1.0`. All other
+resource types are ignored. Only resource types explicitly mentioned will have a weight.
```
Comment (yangwwei) [inline docs/user_guide/sorting_policies.md:154]: I think we need to add a note here, users might be confused about "vcore". We can say something like:
> YuniKorn tracks CPU resources as "vcores" internally, this is mapped to the K8s resource type "cpu". For any other resource types in K8s, the naming is consistent in YuniKorn.

File: docs/user_guide/sorting_policies.md [hunk 2]
```diff
@@ -146,6 +146,33 @@ Resources of the same type are compared for the nodes.
...
+    nodesortpolicy:
+      resourceweights:
+        vcore: 4.0
+        memory: 1.0
+```
+
+With this configuration, In this example, if a node has `90%` of its CPU and `50%` of its memory allocated, the node
+will be considered to be `82%` utilized.
+
+Note that weights are relative to each other, so specifying weights of `{ 4.0, 1.0 }` is equivalent to
```
Comment (yangwwei) [inline docs/user_guide/sorting_policies.md:173]: can we mention negative weight is not allowed?

File: docs/user_guide/sorting_policies.md [hunk 3]
```diff
@@ -146,6 +146,33 @@ Resources of the same type are compared for the nodes.
...
+
+The following configuration entry sets the weight of `vcore` to `4.0` and `memory` to `1.0`. This will weight CPU
+usage four times higher than memory usage:
+```yaml
+partitions:
+  - name: default
+    nodesortpolicy:
+      resourceweights:
+        vcore: 4.0
+        memory: 1.0
```
Comment (yangwwei) [inline docs/user_guide/sorting_policies.md:167]: can we include `type` in this example as well?
I think when user configs `nodesortingpolicy`, type should be always provided.

**Review by yangwwei — CHANGES_REQUESTED** [review]

File: docs/user_guide/sorting_policies.md (hunk 1 above)
Comment (craigcondit) [reply docs/user_guide/sorting_policies.md:154]: Updated.

File: docs/user_guide/sorting_policies.md (hunk 2 above)
Comment (craigcondit) [reply docs/user_guide/sorting_policies.md:177]: Updated.

File: docs/user_guide/sorting_policies.md (hunk 3 above)
Comment (craigcondit) [reply docs/user_guide/sorting_policies.md]: Updated.

**Review by yangwwei — APPROVED** [review]
+1

**Pull request merged**